docs/_ext/sphinx_science/toml.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2023 Science project contributors.
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import dataclasses
+import os
+from collections import deque
+from dataclasses import dataclass
+from enum import Enum
+from types import NoneType
+from typing import Any, Callable, Collection, Iterator, Mapping, Self
+
+from docutils import nodes
+from sphinx_science import directives
+from sphinx_science.directives import bool_option
+from sphinx_science.render import MarkdownParser, Section
+
+from science.dataclass import Dataclass
+from science.dataclass.reflect import (
+    DataclassInfo,
+    FieldInfo,
+    dataclass_info,
+    documented_dataclass,
+)
+from science.frozendict import FrozenDict
+from science.types import TypeInfo
+
+
+@dataclass(frozen=True)
+class TOMLType:
+    label: str
+
+    def render_value(self, value: Any) -> str:
+        return repr(value)
+
+
+@dataclass(frozen=True)
+class PrimitiveType(TOMLType):
+    def render_value(self, value: Any) -> str:
+        if isinstance(value, bool):
+            return "true" if value else "false"
+        return repr(value)
+
+
+@dataclass(frozen=True)
+class ArrayType(TOMLType):
+    @classmethod
+    def create(cls, item_type: TOMLType) -> Self:
+        return cls(label=f"Array[{item_type.label}]", item_type=item_type)
+
+    item_type: TOMLType
+
+    def render_value(self, value: Any) -> str:
+        return repr([self.item_type.render_value(item) for item in value])
+
+
+@dataclass(frozen=True)
+class TableType(TOMLType):
+    @classmethod
+    def create(cls, value_type: TOMLType) -> Self:
+        return cls(label=f"Table[Key, {value_type.label}]", value_type=value_type)
+
+    value_type: TOMLType | None = None
+
+    def render_value(self, value: Any) -> str:
+        if self.value_type:
+            return repr({key: self.value_type.render_value(val) for key, val in value.items()})
+        return repr(value)
```

Comment:
A personal requirement here was to divorce the docs from Python-ness. Science ships as a scie; so the user doesn't know Python, they know TOML, which is the Science lift manifest format. As such, I want the docs to just assume what the tool assumes - which is pretty much just TOML for the config and a subset of shells for when the documenting of the shell-completion comes around.

You can see all of this on the main lift doc here: https://science.scie.app/_/manifest/lift.html from default values to None / empty -> (Optional), etc. There should be no signs of Python-speak.